moveit_simple/src/robot.cpp

```diff
@@ -124,8 +124,6 @@ void Robot::addTrajPoint(const std::string& traj_name, const Eigen::Affine3d pos
                          double time, const InterpolationType& type, const unsigned int num_steps,
                          const std::string& point_name)
 {
-  std::lock_guard<std::recursive_mutex> guard(m_);
```

Comment:
I'm not sure that this is safe to remove since the calls to lookupTransform() can block the thread for up to 5 seconds so multiple threads could produce an indeterminate order of trajectory points. Probably a robot instance shouldn't be shared across multiple threads at all.

Reply:
This PR is intended to get @aaronplusone and @shaun-edwards's feedback and I don't believe that these changes are perfect as is. 

My belief is that we should only use mutex's when we are updating the internal data. If we need a mutex to protect the lookupTransform then the mutex should live in there. It seems strange to me that OnlineRobot does not have an equivalent mutex in it's updateCurrentState subscriber callback. We may need a bigger reassessment of the project's mutex strategy.

Reply:
I'm hesitant to change any of these.  The recursive mutex is admittedly a sign of a poor implementation.  I don't believe we have any tests for the theadsafe-ness of moveit_simple.  Is this a must have change?